src/sentry/integrations/slack/utils.py

```diff
@@ -308,3 +313,56 @@ def build_incident_attachment(incident):
         "color": LEVEL_TO_COLOR["error"],
         "actions": [],
     }
+
+
+# Different list types in slack that we'll use to resolve a channel name. Format is
+# (<list_name>, <result_name>, <prefix>).
+LIST_TYPES = [
+    ("channels", "channels", CHANNEL_PREFIX),
+    ("groups", "groups", CHANNEL_PREFIX),
+    ("users", "members", MEMBER_PREFIX),
+]
+
+
+def get_channel_id(organization, integration_id, name):
+    """
+    Fetches the internal slack id of a channel.
+    :param organization: The organization that is using this integration
+    :param integration_id: The integration id of this slack integration
+    :param name: The name of the channel
+    :return:
+    """
+    name = name.lstrip(strip_channel_chars)
+    try:
+        integration = Integration.objects.get(
+            provider="slack", organizations=organization, id=integration_id
+        )
+    except Integration.DoesNotExist:
+        return None
+
+    token_payload = {"token": integration.metadata["access_token"]}
+
+    # Look for channel ID
+    payload = dict(token_payload, **{"exclude_archived": False, "exclude_members": True})
+
+    session = http.build_session()
+    for list_type, result_name, prefix in LIST_TYPES:
+        # Slack limits the response of `<list_type>.list` to 1000 channels, paginate if
+        # needed
+        cursor = ""
+        while cursor is not None:
+            items = session.get(
+                "https://slack.com/api/%s.list" % list_type,
+                params=dict(payload, **{"cursor": cursor}),
+            )
```

Comment:
What catches 500 errors from slack?

Reply:
Not sure, this is just extracted from the existing slack integration and refactored a little. It seems to work ok at the moment, not sure if we should change it here?

Reply:
I guess we'll find out where the exceptions go next time slack has a bad day.